src/mpi/datatype/typerep/src/typerep_util.h

```diff
@@ -81,71 +59,44 @@
   FLOAT_convert(src, dest) -- converts floating point src into
   external32 floating point format and stores the result in dest.
 
-  BASIC_convert(src, dest) -- converts integral type src into
-  external32 integral type and stores the result in dest.
+  BASIC_convert(const void *src, void *dest, int size) -- converts integral
+  type src into external32 integral type and stores the result in dest.
 
   These two macros compile to assignments on big-endian architectures.
 */
 
-#if (MANUAL_BYTESWAPS == 0)
-#include <netinet/in.h>
-#endif
-
 #define BITSIZE_OF(type)    (sizeof(type) * CHAR_BIT)
 
-#if (MANUAL_BYTESWAPS == 1)
 #define BASIC_convert32(src, dest)      \
 {                                    \
-    dest = (((src >> 24) & 0x000000FF) |\
-            ((src >>  8) & 0x0000FF00) |\
-            ((src <<  8) & 0x00FF0000) |\
-            ((src << 24) & 0xFF000000));\
+    dest = ((((src) >> 24) & 0x000000FF) |\
+            (((src) >>  8) & 0x0000FF00) |\
+            (((src) <<  8) & 0x00FF0000) |\
+            (((src) << 24) & 0xFF000000));\
 }
-#else
-#define BASIC_convert32(src, dest)      \
-{                                    \
-    dest = htonl((uint32_t)src);        \
-}
-#endif
 
-#if (MANUAL_BYTESWAPS == 1)
 #define BASIC_convert16(src, dest)  \
```

Comment:
Maybe define `BASIC_convert16` before `BASIC_convert32` such they appear in order in the source code?

Reply:
I think I did that, then it got lost during fiddling. I'll do that.

Reply:
done.